src/Umbraco.Infrastructure/Migrations/Upgrade/V_17_0_0/MigrateSystemDatesToUtc.cs

```diff
@@ -49,96 +52,153 @@ protected override void Migrate()
+        MigrateDateColumn(scope, "cmsMember", "lastPasswordChangeDate", timeZone);
+        MigrateDateColumn(scope, "umbracoAccess", "createDate", timeZone);
+        MigrateDateColumn(scope, "umbracoAccess", "updateDate", timeZone);
+        MigrateDateColumn(scope, "umbracoAccessRule", "createDate", timeZone);
+        MigrateDateColumn(scope, "umbracoAccessRule", "updateDate", timeZone);
+        MigrateDateColumn(scope, "umbracoCreatedPackageSchema", "updateDate", timeZone);
+        MigrateDateColumn(scope, "umbracoContentVersion", "versionDate", timeZone);
+        MigrateDateColumn(scope, "umbracoContentVersionCleanupPolicy", "updated", timeZone);
+        MigrateDateColumn(scope, "umbracoContentVersionCultureVariation", "date", timeZone);
+        MigrateDateColumn(scope, "umbracoExternalLogin", "createDate", timeZone);
+        MigrateDateColumn(scope, "umbracoExternalLoginToken", "createDate", timeZone);
+        MigrateDateColumn(scope, "umbracoKeyValue", "updated", timeZone);
+        MigrateDateColumn(scope, "umbracoLog", "Datestamp", timeZone);
+        MigrateDateColumn(scope, "umbracoNode", "createDate", timeZone);
+        MigrateDateColumn(scope, "umbracoRelation", "datetime", timeZone);
+        MigrateDateColumn(scope, "umbracoServer", "registeredDate", timeZone);
+        MigrateDateColumn(scope, "umbracoServer", "lastNotifiedDate", timeZone);
+        MigrateDateColumn(scope, "umbracoUser", "createDate", timeZone);
+        MigrateDateColumn(scope, "umbracoUser", "updateDate", timeZone);
+        MigrateDateColumn(scope, "umbracoUser", "emailConfirmedDate", timeZone);
+        MigrateDateColumn(scope, "umbracoUser", "lastLockoutDate", timeZone);
+        MigrateDateColumn(scope, "umbracoUser", "lastPasswordChangeDate", timeZone);
+        MigrateDateColumn(scope, "umbracoUser", "lastLoginDate", timeZone);
+        MigrateDateColumn(scope, "umbracoUser", "invitedDate", timeZone);
+        MigrateDateColumn(scope, "umbracoUserGroup", "createDate", timeZone);
+        MigrateDateColumn(scope, "umbracoUserGroup", "updateDate", timeZone);
 
         scope.Complete();
         Context.Complete();
     }
 
-    private static TimeSpan GetTimezoneOffset(string timeZoneName)
+    private static string GetWindowsTimeZoneId(TimeZoneInfo timeZone)
+    {
+        if (!timeZone.HasIanaId)
+        {
+            return timeZone.Id;
+        }
 
-        // We know the provided timezone name exists, as it's either detected or configured (and configuration has been validated).
-        => TimeZoneInfo.FindSystemTimeZoneById(timeZoneName).BaseUtcOffset;
+        if (!TimeZoneInfo.TryConvertIanaIdToWindowsId(timeZone.Id, out var windowsId))
+        {
+            throw new InvalidOperationException(
+                $"Failed to convert local time zone IANA id '{timeZone.Id}' to a Windows id. Please manually configure the 'Umbraco:CMS:SystemDateMigration:LocalServerTimeZone' app setting with a valid Windows time zone name.");
```

Comment:
```suggestion
                $"Could not update system dates to UTC as it was not possible to convert the detected local time zone IANA id '{timeZone.Id}' to a Windows Id necessary for updates with SQL Server. Please manually configure the 'Umbraco:CMS:SystemDateMigration:LocalServerTimeZone' app setting with a valid Windows time zone name.");
```

Just thought we needed a bit more detail of what's going on when this error is thrown.